tests/L0/run_transformer/run_pipeline_parallel_test.py

```diff
@@ -91,8 +92,10 @@ def forward_backward_func_template(
     tensor_shape[0] = micro_batch_size
 
     update_num_microbatches(0)
-    forward_backward_func(
-        fwd_step_func, batch, model, forward_only=forward_only, tensor_shape=tensor_shape)
+    dtype = torch.half if enable_autocast else None
+    with torch.cuda.amp.autocast():
+        forward_backward_func(
+            fwd_step_func, batch, model, forward_only=forward_only, tensor_shape=tensor_shape, dtype=dtype)
```

Comment:
We can use torch.bfloat16 here as well but autocast of bf16 requires ampere arch.
Notice that `forward_backward_func` takes the keyword only argument of "dtype".